docs/source/validation_rules.rst

```diff
@@ -0,0 +1,714 @@
+  MockFilename,schematic - main/MockFilenameComponent/txt4.txt,syn6168265 # entityId does not exist
+  MockFilename,schematic - main/MockFilenameComponent/txt6.txt,  # missing entityId
+
+
+Rule Combinations
+-----------------
+
+Schematic allows certain combinations of existing validation rules to be used on a single attribute, where appropriate.
+
+Note:  isNa and required can be combined with all rules and rule combos.
+
+Rule combinations: [``list::regex``, ``int::inRange``, ``float::inRange``, ``num::inRange``, ``protectAges::inRange``]
+
+- Format:
+
+  - ``<rule 1> <applicable rule 1 arguments>::<rule 2> <applicable rule 2 arguments>``
+
+  - ``::`` delimiter used to separate each rule
+
+- Example:
+
+  - ``list :: regex search [HTAN][0-9]{1}_[0-9]{4}_[0-9]*``
+
+Component-Based Rule Setting
+----------------------------
+
+**Component-Based Rule Setting** is a powerful feature in data modeling that enables users to create rules tailored to specific subsets of components or manifests. This functionality was developed to address scenarios where a data modeler needs to enforce uniqueness for certain attribute values within one manifest while allowing non-uniqueness in another.
+
+Here's how it works:
+
+1. **Rule Definition at Attribute Level**: Rules are defined at the attribute level within the data model.
+
+2. **Manifest-Level Referencing**: These rules can then be applied (or not) to specific manifests within the data model. This means that rules can be selectively enforced based on the manifest they're associated with.
+
+This feature offers flexibility and applicability beyond its original use case. The new **Component-Based Rule Setting** feature provides users with the following options:
+
+- **Apply a Rule to All Manifests Except Specified Ones**: Users can now define a rule that applies to all manifests within the data model except for those explicitly specified. In cases where exceptions are specified, users have the flexibility to define unique rules for these exceptions or opt not to apply any rule at all.
+
+- **Specify a Rule for a Single Manifest**: Alternatively, users can specify a rule that applies to a single manifest exclusively. This allows for fine-grained control over rule enforcement at the manifest level.
+
+- **Unique Rules for Each Manifest**: Users can also define unique rules for each manifest within the data model. This enables tailored rule enforcement based on the specific requirements and characteristics of each manifest.
+
+By leveraging the enhanced Component-Based Rule Setting feature, data modelers can efficiently enforce rules across their data models with greater precision and flexibility, ensuring data integrity while accommodating diverse use cases and requirements.
+
+Note: All restrictions to rule combos and implementation also apply to component based rules.
+
+Note: As always try the rule combos with mock data to ensure they are working as intended before using in production.
```

Comment:
Do you think it would make sense to highlight the "Note" section using a note code block? From my perspective, that makes the notes stand out more and easier to notice.